connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala

```diff
@@ -2909,6 +2918,36 @@ class SparkConnectPlanner(val sessionHolder: SessionHolder) extends Logging {
         session.streams.resetTerminated()
         respBuilder.setResetTerminated(true)
 
+      case StreamingQueryManagerCommand.CommandCase.ADD_LISTENER =>
+        val listenerPacket = Utils
+          .deserialize[StreamingListenerPacket](
+            command.getAddListener.getListenerPayload.toByteArray,
+            Utils.getContextOrSparkClassLoader)
+        val listener: StreamingQueryListener = listenerPacket.listener
+          .asInstanceOf[StreamingQueryListener]
+        val id: String = listenerPacket.id
+        sessionHolder.cacheListenerById(id, listener)
+        session.streams.addListener(listener)
```

Comment:
So, from this point on this listener is treated as legacy StreamingListener. I.e. it is not a Connect streaming listener. Is that fine? Wondering what happens if user tries to run a spark command inside the listener (added a comment about this in unit test).
For the first version here it might be fine, but I think we should have clearly defined TODO for this improvement. cc: @WweiL

Reply:
Are you talking about the cache we introduced here? We'll have to do this for `removeListener()` to work properly.

Reply:
I see, will add a TODO for that. What do we do for Python Listener @WweiL?

Reply:
Python is adding a new method on the listener to get the session (which would be a connect remote session). 
This API might change.

Reply:
Added a [TODO](https://github.com/apache/spark/pull/41752/commits/5ef8b52ef2f61c2cd5202cd5dabcde9df3c57528) for this. Also created a Jira [SPARK-44400](https://issues.apache.org/jira/browse/SPARK-44400). I might not have the permission for our [streaming spark connect epic](https://issues.apache.org/jira/browse/SPARK-42938), not able to add the new task under it.

Reply:
Thanks. I moved it to the Epic.